src/briefcase/platforms/linux/appimage.py

```diff
@@ -142,41 +142,85 @@ class LinuxAppImageBuildCommand(LinuxAppImageMixin, BuildCommand):
     description = "Build a Linux AppImage."
 
     def verify_tools(self):
+        """Verify that the AppImage linuxdeploy tool and plugins exist.
+
+        :param app: The application to build
+        """
         super().verify_tools()
         self.linuxdeploy = LinuxDeploy.verify(self)
 
-    def build_app(self, app: BaseConfig, **kwargs):
+    def build_app(self, app: AppConfig, **kwargs):
         """Build an application.
 
         :param app: The application to build
         """
-        self.logger.info("Building AppImage...", prefix=app.app_name)
+        # Build a dictionary of environment definitions that are required
+        env = {}
+
+        self.logger.info("Checking for Linuxdeploy plugins...", prefix=app.app_name)
+        try:
+            plugins = self.linuxdeploy.verify_plugins(
+                app.linuxdeploy_plugins,
+                bundle_path=self.bundle_path(app),
+            )
 
+            self.logger.info("Configuring Linuxdeploy plugins...", prefix=app.app_name)
+            # We need to add the location of the linuxdeploy plugins to the PATH.
+            # However, if we arerunning inside Docker, we need to know the
+            # environment *inside* the Docker container.
+            if self.use_docker:
+                echo_cmd = ["/bin/bash", "-c", "echo $PATH"]
+                base_path = self.Docker(self, app).check_output(echo_cmd).strip()
```

Comment:
This can bomb out if the Docker image disappears.

```
[helloworld] Configuring Linuxdeploy plugins...

>>> Running Command:
>>>     docker run --volume /home/russell/tmp/beeware/helloworld/linux:/app:z --volume /home/russell/.cache/briefcase:/home/brutus/.cache/briefcase:z --rm briefcase/com.example.helloworld:py3.10 /bin/bash
-c 'echo $PATH'
Unable to find image 'briefcase/com.example.helloworld:py3.10' locally
docker: Error response from daemon: pull access denied for briefcase/com.example.helloworld, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
See 'docker run --help'.
>>> Return code: 125

Log saved to /home/russell/tmp/beeware/helloworld/briefcase.2022_07_21-14_19_42.build.log

Traceback (most recent call last):
  File "/home/russell/github/danyeaw/briefcase/venv-3.10/bin/briefcase", line 33, in <module>
    sys.exit(load_entry_point('briefcase', 'console_scripts', 'briefcase')())
  File "/home/russell/github/danyeaw/briefcase/src/briefcase/__main__.py", line 15, in main
    command(**options)
  File "/home/russell/github/danyeaw/briefcase/src/briefcase/commands/build.py", line 58, in __call__
    state = self._build_app(
  File "/home/russell/github/danyeaw/briefcase/src/briefcase/commands/build.py", line 39, in _build_app
    state = self.build_app(app, **full_options(state, options))
  File "/home/russell/github/danyeaw/briefcase/src/briefcase/platforms/linux/appimage.py", line 173, in build_app
    base_path = self.Docker(self, app).check_output(echo_cmd).strip()
  File "/home/russell/github/danyeaw/briefcase/src/briefcase/integrations/docker.py", line 324, in check_output
    return self._subprocess.check_output(
  File "/home/russell/github/danyeaw/briefcase/src/briefcase/integrations/subprocess.py", line 297, in check_output
    cmd_output = self._subprocess.check_output(
  File "/home/russell/.pyenv/versions/3.10.4/lib/python3.10/subprocess.py", line 420, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/home/russell/.pyenv/versions/3.10.4/lib/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['docker', 'run', '--volume', '/home/russell/tmp/beeware/helloworld/linux:/app:z', '--volume', '/home/russell/.cache/briefcase:/home/brutus/.cache/briefcase:z', '--rm', 'briefcase/com.example.helloworld:py3.10', '/bin/bash', '-c', 'echo $PATH']' returned non-zero exit status 125.
```

Reply:
I hit this error a couple of times, but I put it down to the fact that I was doing some "off script" rebuilds and cache refreshes. Definitely worth addressing, but I don't think it's a blocker for this PR specifically.